### DISCUSSION TITLE
splitting of path name

### DISCUSSION THREAD

**Discussion by bereketdora** [jddingemanse/awtiCodeDev#22] [Q&A] [answered]
I tried to separate the DD/MM/YY format and take the monthly mean of the precipitation data. I've set myself the task of overcoming this, though. please, Can somebody assist me?

**Comment by jddingemanse** [comment, marked as answer]
> I tried to separate the DD/MM/YY format and take the monthly mean of the precipitation data. I've set myself the task of overcoming this, though. please, Can somebody assist me?

I hope I understand your question correctly; I assume you have a column with dates (and next to that a column with precipitation values). If that is not the case, please explain what you have, you can for example show a small screen shot.

**Option 1: getting time information from a timestamp column**
Load your data as a Pandas DataFrame, and turn your column with dates into a Pandas Timestamp with `pd.to_datetime()`. Once you have a column with actual Timestamps, you can obtain parts of the Timestamp with the built-in method `.dt`. For example, if you have a dataframe `df`, and a newly created column `df.timestamp` with those Timestamps, you can get a column with only month numbers with the command `df.timestamp.dt.month`, or a column with only years with the command `df.timestamp.dt.year`.

**Option 2: using string methods**
Another way to split time data of a column, is by actualy using them as strings. Load your data as Pandas DataFrame, and keep a column with the dates **as strings**. To any column with strings, you can access string methods (and do that method on all strings at the same time) through the built-in method .str. For example, if you have strings with all format _DD/MM/YY_ in a column `df.dates`, you can select the MM/YY part with `df.dates.str.slice(start=3)`, and it will give you a column with all date strings starting from the fourth character (so, without 'DD/'. 

**Monthly averages with the method `.resample()`**
Apart from that, if indeed you have a column with dates, and want to take monthly averages, I think it is most easy to work with the method `.resample()`. For that, you will need to load your data as Pandas DataFrame, and turn your column with date strings into Timestamps with `pd.to_datetime()`. Then, assuming you have a dataframe `df` with a column `df.timestamp` with those newly created timestamps, you can use `df.resample(rule='M',on='timestamp').mean()` to obtain monthly (**rule='M'**) averages (**.mean()**) based on those timestamps (**on='timestamp'**).

**Reply by bereketdora** [reply]
thank you very much for your brief explanation to solve my problem.

**Comment by bereketdora** [comment]
Many thanks.

ᐧ

**Comment by bereketdora** [comment]
Dear Johannes,
I hope everything is good with you. I have some questions regarding to
final assignment.   Do you have time to answer my questions please?  if you
have more time please visit the following one show the ways.
[image: image.png]
now I want to add row at the end and to get the sum of of each station.
ᐧ

**Comment by bereketdora** [comment]
I'm sorry, I understand. thanks
ᐧ

**Reply by jddingemanse** [reply]
I think you mean you found the answer to your own question? If so, please explain what you found, so that others can learn from it. Thanks in advance!
